python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -650,6 +669,31 @@ def _get_entities_from_payload(self, data):
         else:
             raise RuntimeError("Unable to determine an entity from data: %s" % data)
 
+    def _get_entity_type_whitelist(self):
+        """
+        Gets a set of entity types that are supported by the browser
+        integration. This set is built from a list of constant entity
+        types, plus all entity types that a PublishedFile entity is
+        allowed to link to, per the current site's schema.
+
+        :returns: A set of string entity types.
+        :rtype: set
+        """
+        if self.ENTITY_TYPE_WHITELIST not in self._cache:
+            type_whitelist = constants.BASE_ENTITY_TYPE_WHITELIST
```

Comment:
TL;DR - There's a common set for the whole site, and some entities can be masked on a per-project basis.

Boring details:

The list of possibles entity types to a given field is determined by the data_type_properties property of the properties field of a DisplayColumn. (still with me?)

```
DisplayColumn
   - properties
      - data_type_properties
         - allowed_entity_types
```

(there's a special case where the allowed_entity_types isn't set on the DisplayColumn instance itself but determined by the field's Rails association, but I digress)

The display columns are configured at the entity level (site-wide) not at a site level. 

Now,  there are UIVisibilityRules which can be set that makes a given entity type visible or not in a given project. 

Invoking schema_field_read takes into account both the site-wide allowed entity types _and_ the per-project UIVisibilityRules. 

For reference:

https://github.com/shotgunsoftware/shotgun/blob/develop/7.1/app/apis/api3_handler.rb#L1272
https://github.com/shotgunsoftware/shotgun/blob/develop/7.1/app/apis/api3_handler.rb#L3511
https://github.com/shotgunsoftware/shotgun/blob/develop/7.1/app/models/display_column.rb#L534